staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -695,6 +695,10 @@ func (s *store) List(ctx context.Context, key string, opts storage.ListOptions,
 			break
 		}
 		key = string(lastKey) + "\x00"
+		if fromRV == nil {
```

Comment:
if we're using ResourceVersionMatchNotOlderThan, a non-nil fromRV doesn't mean we added a WithRev option.

I think we should track whether we added WithRev explicitly, and make that the condition here.

Reply:
That also means we need a test that demonstrates paging+filtering+ResourceVersionMatchNotOlderThan that would exercise this

Reply:
Good catch, test added and test is already catching it.

Now I need to fix the code.

Reply:
Fixed